windows/src/global/delphi/general/Upload_Settings.pas

```diff
@@ -54,7 +54,12 @@ interface
   URLPath_Community = '/go/'+SKeymanVersion+'/community';
 
   // Keyboard download and installation
-  URLPath_RegEx_MatchKeyboardsInstall = '^http(?:s)?://[^/]+/keyboards/install/([^?/]+)(?:\?(.+))?$';
+  URLPath_RegEx_MatchKeyboardsInstall = '^http(?:s)?://keyman(?:-staging)?\.com(?:\.local)?/keyboards/install/([^?/]+)(?:\?(.+))?$';
+                                  // e.g. https://keyman.com/keyboards/install/foo
+  UrlPath_RegEx_MatchKeyboardsRoot = '^http(?:s)?://keyman(?:-staging)?\.com(?:\.local)?/keyboards([/?].*)?$';
+                               // e.g. http://keyman.com.local/keyboards/foo
+  UrlPath_RegEx_MatchKeyboardsGo = '^http(?:s)?://keyman(?:-staging)?\.com(?:\.local)?/go/windows/[^/]+/download-keyboards';
```

Comment:
Does the regex need to go all the way to the end (like _MatchKeyboardsRoot)?
`([/?].*)?$`

Reply:
It's not terribly important; it probably should...